decorator.rb

```diff
@@ -0,0 +1,10 @@
+require_relative 'nameable'
+
+class Decorator < Nameable
+  attr_accessor :nameable
+
+  def initialize(nameable)
+    super()
+    @nameable = nameable
+  end
+end
```

Comment:
- [ Required ] I think it would be nice if the `Decorator` class overrides the `correct_name` method for your application to work as expected. Currently, the application gives the error below when it's run. I would like to suggest the code below as a way to solve this problem.

## Example code
```ruby
require_relative 'nameable'

class Decorator < Nameable
  attr_accessor :nameable

  def initialize(nameable)
    super()
    @nameable = nameable
  end

  def correct_name
    @nameable.correct_name
  end
end
```

## Error when running the App
![Screenshot from 2023-10-12 00-18-08](https://github.com/ahmidou09/OOP-school-library/assets/16392046/a366a5aa-8e54-4fcd-8ca3-f09645dc7177)